pkg/storegateway/series_chunks.go

```diff
@@ -58,8 +60,13 @@ func newSeriesChunksSeriesSet(from seriesChunksSetIterator) storepb.SeriesSet {
 	}
 }
 
-func newSeriesSetWithChunks(ctx context.Context, chunkReaders chunkReaders, chunksPool pool.Bytes, batches seriesChunkRefsSetIterator, stats *safeQueryStats) storepb.SeriesSet {
-	return newSeriesChunksSeriesSet(newPreloadingSetIterator[seriesChunksSet](ctx, 1, newLoadingSeriesChunksSetIterator(chunkReaders, chunksPool, batches, stats)))
+func newSeriesSetWithChunks(ctx context.Context, chunkReaders chunkReaders, chunksPool pool.Bytes, batches seriesChunkRefsSetIterator, stats *safeQueryStats, metrics *BucketStoreMetrics) storepb.SeriesSet {
+	var iterator seriesChunksSetIterator
+	iterator = newLoadingSeriesChunksSetIterator(chunkReaders, chunksPool, batches, stats)
+	iterator = newDurationMeasuringIterator[seriesChunksSet](iterator, metrics.iteratorLoadDurations.WithLabelValues("chunks_load"))
+	iterator = newPreloadingSetIterator[seriesChunksSet](ctx, 1, iterator)
+	iterator = newDurationMeasuringIterator[seriesChunksSet](iterator, metrics.iteratorLoadDurations.WithLabelValues("chunk_preloaded"))
```

Comment:
[nit] Let's be consistent with `chunks_load`.

```suggestion
	iterator = newDurationMeasuringIterator[seriesChunksSet](iterator, metrics.iteratorLoadDurations.WithLabelValues("chunks_preloaded"))
```